### ISSUE TITLE
No reconnect after a "short stop" MINI ID 5

### ISSUE THREAD

**Issue by Vasmares** [BimmerGestalt/AAIdrive#438]
Hi,
I have monitored this that last days, and it seems the app / the connection does not like short stops.

I left work by car and everything connected as expected and worked fine.
After a ~20min. drive I stopped at a store to get stuff. This took only like 2-3 min.
After getting back into the car, the connection between my car and the app seemed to have problems.

ID5 showed me, that my phone was connected (bluetooth) and the "connected apps" where working.
Your app also showed that the connection was up and running.

But there where no apps loaded. Like for example spotify, or even the menu item to adjust the notification settings.
Only after I arrived at home after another ~10min drive and unlocked my phone to check the app the connection slowly "came back to live". But not as fast as it does when you first set off in the morning or so...

Do you get what I am trying to say ? ^^
I am not sure this is your apps "fault", or maybe a bug in ID5 for MINI

I attached some screenshots, hope this gets my problem across...

![photo_2021-10-14_22-47-13](https://user-images.githubusercontent.com/10863304/137393383-be1e2c2e-34ee-4352-be94-9dd60d9b6fdb.jpg)
![photo_2021-10-14_22-47-21](https://user-images.githubusercontent.com/10863304/137393386-169d3d0e-981a-4664-b07f-0c714993c48e.jpg)
![photo_2021-10-14_22-47-27](https://user-images.githubusercontent.com/10863304/137393388-0c24e0dd-1095-4708-977f-b721443bede5.jpg)
![photo_2021-10-14_22-47-32](https://user-images.githubusercontent.com/10863304/137393390-30cc29aa-ca5c-4fc5-92e6-ead8bc08eaf5.jpg)
![photo_2021-10-14_22-47-35](https://user-images.githubusercontent.com/10863304/137393391-10cb64a8-f684-40b9-b761-f788d1ae3423.jpg)
![photo_2021-10-14_22-47-39](https://user-images.githubusercontent.com/10863304/137393393-e6683a99-f41b-4c5f-9636-6bae32faaedd.jpg)

**Comment by hufman** [comment]
That's very curious! (Excellent taste on picking the Mini Clubman btw)
The MainService should be triggered by multiple entrypoints (Bluetooth connection, CarProber notices the port is available, MyBMW/Mini announces the connection and NotificationListener hears it), and it should automatically launch the car app modules when the connection is detected. So, when you opened the app to check on it, it should launch the apps if they weren't running. Did you say that the apps stayed unavailable even while the phone UI shows the green connection?
The green connection does technically reflect the overall car connection status, and not whether the app modules are actually running. However, the apps should still be launched anyways when opening the phone UI.

The short stop into the store is tricky too. The MainService does shut down after 2 minutes (if the car's Bluetooth was still connected when the shutdown timer was scheduled), but it should start back up again as soon as the Bluetooth connection is resumed. Has it only happened this one time, or how often do you estimate it has happened to you?

**Comment by Vasmares** [comment]
This seems to be fixed with the nightly build from last friday.
But now another issue is at hand.

New Issue opened here: https://github.com/BimmerGestalt/AAIdrive/issues/463
